Advanced upload page

Seneli opened the pull request:
## This PR addresses the following milestones/tasks:
- 5.3 #3 

## Things not included in this PR:
- Logos are not added in since the code for adding in font-awesome is in the dashboard PR.
- This PR does everything except send the final request to the back end, but the function that will do that has been identified with a comment. 
- If we have time (unlikely) we could animate the components in this PR but it's not necessary. 

## Summary of changes file by file:

- `Upload.tsx` - the component that encapsulates everything shown in the upload page. It is the most high level component in this PR.

- `CarouselController.tsx` is the left hand side of the component thingy shown below. `CarouselBody.tsx` is the right side of the component thingy shown below.
![Screen Shot 2023-01-23 at 3 16 53 PM](https://user-images.githubusercontent.com/44822892/214140975-c1405d3d-975c-4844-8f9b-83fd9c6948f2.png)
 
- `UploadStep.tsx` - the entirety of one of the upload steps. It's a wrapper for the radio dropdowns or file upload component listed right after this :) 
![Screen Shot 2023-01-23 at 3 16 11 PM](https://user-images.githubusercontent.com/44822892/214140840-39a4d61b-3507-4b95-8170-66b060a9b6d6.png)

- `RadioDropdown.tsx` - has a list of dropdown options that you can choose from
![Screen Shot 2023-01-23 at 3 15 47 PM](https://user-images.githubusercontent.com/44822892/214140748-78b25855-471e-4df0-bd81-d337104aff5d.png)

- `FileUpload.tsx` - component that handles the upload of the file. 
![Screen Shot 2023-01-23 at 3 15 33 PM](https://user-images.githubusercontent.com/44822892/214140710-931100cb-b032-44f2-b553-d1ca3388f62a.png)

Seneli committed:
basic dropdown menu template done

Seneli committed:
super messy unanimated carousel with no timeline component

Seneli committed:
added state management to upload page

Seneli committed:
removed half finished upload function for now

Seneli committed:
made it a bit nicer to look at for the demo

z-arnott commented:
can you remove "input manually" 
I don't see a text box entry yet - and we decided as a group during design that this adds complexity without much benefit

and remove step 2 completely

z-arnott commented:
change 'submit query' to 'submit' in Step 3, pictured below
![image](https://user-images.githubusercontent.com/50159450/215290606-129dbe12-d09d-4ec6-91d3-e9c07021a6c1.png)

z-arnott commented:
what I don't see in the UI or in the code (I might have missed it) is a way to determine the filetype 
it would be simplest to have the user specify

z-arnott commented on client/src/components/FileUpload.tsx:
```diff
@@ -0,0 +1,88 @@
+  const [file, setFile] = useState();
+  const [uploadMessage, setUploadMessage] = useState('');
+
+  const setUploadStateMessage = () => {
```
how are you determining filetype?

Seneli committed:
added automatic step changes and limited accepted file types

Seneli committed:
changed step2 to file type and added axios

Seneli committed:
issues with file upload but template is there

Seneli commented:
I need some help with this one. When I console.log the request I thought the query params would at least send, but that and the files don't make it to the back end. I'm not sure why but if someone else can debug this issue for now it would help me move on :") 

![Screen Shot 2023-01-31 at 7 43 24 AM](https://user-images.githubusercontent.com/44822892/215763049-b341e7fb-d736-46d1-9884-ebc3f7c4cd04.png)

![Screen Shot 2023-01-31 at 7 44 16 AM](https://user-images.githubusercontent.com/44822892/215763250-402a2d5c-1676-4a4e-bc02-3978434adff7.png)

z-arnott commented:
I was able to  send requests with the following hardcoded values  in FIleUpload.tsx
![image](https://user-images.githubusercontent.com/50159450/216209200-f063fd7f-d035-425c-bd2d-d04c11b0eaac.png)

Seneli committed:
merged main

Seneli commented:
This is the error I am getting now. I think I am able to send things well but I couldn't figure out how to resolve this. 

![Screen Shot 2023-02-04 at 3 00 40 PM](https://user-images.githubusercontent.com/44822892/216787365-937eb83b-b7a8-4f0f-a90d-e0394659f83a.png)

Seneli committed:
hard coded query parameters for file upload

Seneli committed:
added spdx tag value option and state handling

Seneli committed:
added session id generator logic

Seneli committed:
file type validation - best i got for rn

Seneli committed:
added icons

Seneli committed:
used regex as preliminary input validation

Seneli committed:
updated error message

Seneli commented:
I think this is going to be my final commit to the upload page so here's my changes since we talked last time :) 

#### 1. Added a function to upload the file to the back end. 
- There's some kinda vague error reporting  to this and file type validation. You can test it out to see more of what that looks like.
- Currently, input validation is done through getting the file type extension via regex. It's not perfect but it's something.
![Screen Shot 2023-02-04 at 6 36 36 PM](https://user-images.githubusercontent.com/44822892/216794019-a3c07251-fcbd-4cea-add9-a5a1e9729253.png)
![Screen Shot 2023-02-04 at 6 37 09 PM](https://user-images.githubusercontent.com/44822892/216794036-c028f861-7ebd-44af-8227-a156c7ed367f.png)



#### 2. Added option to select that you wanted to upload a SPDX document in the tag value.
![Screen Shot 2023-02-04 at 6 35 11 PM](https://user-images.githubusercontent.com/44822892/216793974-ba14ca72-a554-48ed-8683-63202e1151df.png)
- This also included making it so whenever a user changed their file format they had to re-specify their file type, since the options between the 2 aren't the same. 

#### 3. Added icons to the left side
![Screen Shot 2023-02-04 at 6 17 31 PM](https://user-images.githubusercontent.com/44822892/216793583-4a9d0e99-9f4f-411e-803e-19a8bf485022.png)

z-arnott reviewed (approved):
Looks great - thanks for making those changes
Tested it with back end integration today and it's working great :)
